core/chains/evm/txmgr/eth_confirmer.go

```diff
@@ -1365,25 +1368,27 @@ func saveConfirmedMissingReceiptAttempt(q pg.Q, lggr logger.Logger, attempt *Eth
 	return errors.Wrap(err, "saveConfirmedMissingReceiptAttempt failed")
 }
 
-func saveSentAttempt(q pg.Queryer, lggr logger.Logger, attempt *EthTxAttempt, broadcastAt time.Time) error {
+func saveSentAttempt(q pg.Queryer, timeout time.Duration, lggr logger.Logger, attempt *EthTxAttempt, broadcastAt time.Time) error {
 	if attempt.State != EthTxAttemptInProgress {
 		return errors.New("expected state to be in_progress")
 	}
 	attempt.State = EthTxAttemptBroadcast
-	return errors.Wrap(saveAttemptWithNewState(q, lggr, *attempt, broadcastAt), "saveSentAttempt failed")
+	return errors.Wrap(saveAttemptWithNewState(q, timeout, lggr, *attempt, broadcastAt), "saveSentAttempt failed")
 }
 
-func saveInsufficientEthAttempt(q pg.Queryer, lggr logger.Logger, attempt *EthTxAttempt, broadcastAt time.Time) error {
+func saveInsufficientEthAttempt(q pg.Queryer, timeout time.Duration, lggr logger.Logger, attempt *EthTxAttempt, broadcastAt time.Time) error {
 	if !(attempt.State == EthTxAttemptInProgress || attempt.State == EthTxAttemptInsufficientEth) {
 		return errors.New("expected state to be either in_progress or insufficient_eth")
 	}
 	attempt.State = EthTxAttemptInsufficientEth
-	return errors.Wrap(saveAttemptWithNewState(q, lggr, *attempt, broadcastAt), "saveInsufficientEthAttempt failed")
+	return errors.Wrap(saveAttemptWithNewState(q, timeout, lggr, *attempt, broadcastAt), "saveInsufficientEthAttempt failed")
 
 }
 
-func saveAttemptWithNewState(q pg.Queryer, lggr logger.Logger, attempt EthTxAttempt, broadcastAt time.Time) error {
-	return pg.SqlxTransactionWithDefaultCtx(q, lggr, func(tx pg.Queryer) error {
+func saveAttemptWithNewState(q pg.Queryer, timeout time.Duration, lggr logger.Logger, attempt EthTxAttempt, broadcastAt time.Time) error {
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
```

Comment:
Hmmm... we're trying to eliminate all instances of `context.Background()`, as it can result in non-graceful shutdowns.

I guess the real problem though is we ignore this arg in `Start()` rather than saving a context:
```
func (ec *EthConfirmer) Start(_ context.Context) error {
```
If there's an easy way to save a context here and pass it along, that would be better.. but I guess it's technically out of scope of this PR.

Reply:
These are not actually changing behavior - just inlining the original `SqlxTransactionWithDefaultCtx` with the timeout from config, instead of using the global var internally.

That being said, `context.Background()` is not necessarily a problem and is actually often inevitable, for example [utils.ContextFromChan](https://github.com/smartcontractkit/chainlink/blob/02837885996b1efd7730c9f78608641879e20a24/core/utils/utils.go#L437-L438) uses it internally. Also we should _never_ be saving the context from a `Start` method inside a service. The context passed to `Start` has a different meaning - e.g. it may have an arbitrary timeout to give up on a slow startup. We shouldn't even be storing contexts in structs at all. There are a few exceptions, but generally only for API compatibility. From https://pkg.go.dev/context

> Do not store Contexts inside a struct type; instead, pass a Context explicitly to each function that needs it.

I agree there is room for improvement here, at least for readability (I would like to see use follow the standard patterns of explicit ctx param first), but I don't fully grok how the `Queryer` is being used in this case (it can also carry a ctx/timeout) and refactoring this system seems out of scope for this PR. Feel free to create a story though.